lib/vtls/vtls.c

```diff
@@ -1243,7 +1243,7 @@ static size_t Curl_multissl_version(char *buffer, size_t size)
 
     selected = current;
 
-    for(i = 0; available_backends[i]; i++) {
+    for(i = 0; available_backends[i] && p < backends + sizeof(backends); i++) {
```

Comment:
You'll have to account for each time p is incremented. How about
~~~c
  if(current != selected) {
    char *p = backends;
    char *end = backends + sizeof(backends);
    int i;

    selected = current;

    for(i = 0; available_backends[i] && p < end - 3; i++) {
      if(i)
        *(p++) = ' ';
      if(selected != available_backends[i])
        *(p++) = '(';
      p += available_backends[i]->version(p, end - p - 3);
      if(selected != available_backends[i])
        *(p++) = ')';
    }
    *p = '\0';
    total = p - backends;
  }
~~~
Also I suggest increase it to like 1k size, 200 seems low for multiple versions.

Reply:
also what is supposed to happen here if(current==selected)?

Reply:
If we need to increase the `backends` buffer, or perhaps independently of that, we should also increase the `version` and the `ssl_version` buffers that are used to get this output (in lib/version.c) - both of them are probably in the smaller end if someone builds a curl with all possible TLS backends enabled (not sure exactly how many and which that might be).

Reply:
> also what is supposed to happen here if(current==selected)?

It's keeping the buffer cached in case the current TLS backend hasn't changed between calls, as the selectdt backend is printed differently from the rest.

Reply:
> You'll have to account for each time p is incremented.

Good catch, fixing.

> Also I suggest increase it to like 1k size, 200 seems low for multiple versions.

I'm not sure 200 characters is too low, and 1k seems like excessive stack usage, but there are definitely buffers in the version printing which needs to get bumped. That's will be a separate PR though.

Reply:
I would imagine 200 should be enough. If you manage to build with **6** backends, that makes it 33 bytes per backend, and I don't think any backend uses that much. **8** backends give 25 bytes each, probably also plenty enough.

I doubt we can build with more than 8, if even that...

Reply:
> Good catch, fixing.

I've added `p += available_backends[i]->version(p, end - p - 2);` since it wasn't in the fix. Technically if bytes written returned is always going to be less than the count passed (eg version(foo, 5) and therefore the max it will return is 4 bec it accounts for the null) you only need to subtract 1 but I think subtracting 2 makes the intention clearer.